File: ProblemSolve/268.cpp
/******************
 * Missing Number *
 ******************/

class Solution {
public:
    int missingNumber(vector<int>& nums) {        
        int len = nums.size();        
        sort(nums.begin(), nums.end());
                
        //정렬 후에 idx에 맞는 숫자가 아니면 리턴
        for(int i=0; i<len; i++){
            if(nums[i] != i){
                return i;
            }
        }
        
        return len;
    }
};

File: ProblemSolve/268.cs
/******************
 * Missing Number *
 ******************/

public class Solution {
    public int MissingNumber(int[] nums) {
        bool isContainZero = false;
        
        for(int i=0; i<nums.Length; ++i){
            if(nums[i] == 0){
                isContainZero = true;
            }
            else{
                int originalVal = Math.Abs(nums[i]);
                if(originalVal == nums.Length + 1){
                    isContainZero = true;
                    continue;
                }
                
                nums[originalVal - 1] *= -1;
                if(nums[originalVal - 1] == 0){
                    nums[originalVal - 1] = -nums.Length - 1;
                }
            }
        }
        
        if(!isContainZero){
            return 0;
        }
        
        for(int i=0; i<nums.Length; ++i){
            if(nums[i] >= 0){
                return i + 1;
            }
        }
        
        return 0;
    }
}